families/block_info/rust/sawtooth_block_info/src/handler.rs

```diff
@@ -0,0 +1,306 @@
+                        "Timestamp must be greater than previous block's. Got {},
+                         expected {}",
+                        next_block.get_timestamp(),
+                        prev_block.get_timestamp()
+                    )));
+                }
+
+                let mut deletes: Vec<String> = Vec::new();
+                let mut sets: Vec<(String, Vec<u8>)> = Vec::new();
+
+                // Compute  deletes
+                config.set_latest_block(next_block.get_block_num());
+                while config.get_latest_block() - config.get_oldest_block()
+                    > config.get_target_count()
+                {
+                    deletes.push(create_block_address(config.get_oldest_block()));
+                    let oldest_block = config.get_oldest_block();
+                    config.set_oldest_block(oldest_block + 1);
+                }
+
+                // Compute sets
+                let mut sets = HashMap::new();
+                sets.insert(get_config_addr(), serialize_protobuf(&config)?);
+                sets.insert(
+                    create_block_address(next_block.get_block_num()),
+                    serialize_protobuf(next_block)?,
+                );
+
+                (sets, deletes)
+            }
+            _ => {
+                // If target count or sync tolerance were not specified in the
+                // txn, use default values.
+                config.set_target_count(if target_count != 0 {
+                    target_count
+                } else {
+                    DEFAULT_TARGET_COUNT
+                });
+                config.set_sync_tolerance(if sync_tolerance != 0 {
+                    sync_tolerance
+                } else {
+                    DEFAULT_SYNC_TOLERANCE
+                });
+                config.set_latest_block(next_block.get_block_num());
+                config.set_oldest_block(next_block.get_block_num());
+
+                validate_timestamp(next_block.get_timestamp(), config.get_sync_tolerance())?;
+
+                let config_bytes = serialize_protobuf(&config)?;
+                let block_bytes = serialize_protobuf(next_block)?;
+
+                let mut sets = HashMap::new();
+                sets.insert(get_config_addr(), config_bytes);
+                sets.insert(
+                    create_block_address(next_block.get_block_num()),
+                    block_bytes,
+                );
+
+                (sets, Vec::new())
+            }
+        };
+
+        if !deletes.is_empty() {
+            context.delete_state(deletes.to_vec())?;
+        }
+
+        if !sets.is_empty() {
+            context.set_state(sets)?;
+        }
+
+        Ok(())
+    }
+}
+
+#[cfg(target_arch = "wasm32")]
+// Sabre apply must return a bool
+fn apply(request: &TpProcessRequest, context: &mut TransactionContext) -> Result<bool, ApplyError> {
+    let handler = BlockInfoTransactionHandler::new();
+    match handler.apply(request, context) {
```

Comment:
You could just use `.map(|_| true)`, instead of the match statement here.

Reply:
I basically  copied this `apply` method from the [Sabre documentation](https://sawtooth.hyperledger.org/docs/sabre/nightly/master/application_developer_guide.html#converting-a-transaction-processor-to-a-smart-contract). Would it be better to keep it as it is in the documentation, for clarity? I also realize the documentation might change